components/CompletionScoreCard.tsx

```diff
@@ -0,0 +1,213 @@
+'use client';
+
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+import FiberManualRecordIcon from '@mui/icons-material/FiberManualRecord';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import Chip from '@mui/material/Chip';
+
+interface CompletionScoreCardProps {}
+
+interface MockData {
+  title: string;
+  linkText: string;
+  linkHref: string;
+  scoreData: scoreData[];
+}
+
+interface scoreData {
+  status: 'High' | 'Medium' | 'Low';
+  total: number;
+  completed: number;
+}
+
+const mockData = {
+  title: 'Zero COventry',
+  linkText: 'netzerocities.kausal.tech/coventry-netzero',
+  linkHref: '#',
+  scoreData: [
+    { status: 'High', total: 13, completed: 10 } as scoreData,
+    { status: 'Medium', total: 10, completed: 7 } as scoreData,
+    { status: 'Low', total: 8, completed: 2 } as scoreData,
+  ],
+};
+
+const calculateScores = (scoreData: scoreData[]) => {
+  return scoreData.map((data) => ({
+    status: data.status,
+    score: `${data.completed}/${data.total}`,
+  }));
+};
+
+const calculatePercentage = (scoreData: scoreData[]) => {
+  const totalCompleted = scoreData.reduce(
+    (acc, curr) => acc + curr.completed,
+    0
+  );
+  const totalItems = scoreData.reduce((acc, curr) => acc + curr.total, 0);
+  return totalItems === 0 ? 0 : (totalCompleted / totalItems) * 100;
+};
```

Comment:
This is where we could add weighting to the score based on status. For example, completing ten `High` measures would have a higher impact on the score than completing ten `Medium` or `Low` measures.

I'm open to ideas on how we could do this, but one approach could be to first calculate the percentages per status and then give them weights when calculating the overall percentage. For example, we would:

1. Calculate the percentage per status: `High (10/13 = 0.77)`, `Medium (7/10 = 0.7)`, `Low (2/8 = 0.25)`
2. Decide the "weighting value" for each status: `High = 60% of the total`, `Medium = 30% of the total`, `Low = 10% of the total` (these must add up to 100%)
3. Multiply the percentage from step 1 with the weighting value and sum the numbers: `(60*0.77) + (30*0.7) + (10*0.25) = 70%`

How does that sound? We could make the "weighting value" for each score a constant so its easy to change later